client/src/partials/reports/global_transaction/global_transaction.js

```diff
@@ -133,45 +142,64 @@ angular.module('bhima.controllers')
     }
 
     function search () {
-      if (!$scope.model.account_id) { return; }
-      $scope.mode = $scope.model.account_id !== 0 ? 'selected' : 'all';
+      $scope.mode = ($scope.model.account_id && $scope.model.account_id > 0) ? 'selected' : 'all';
```

Comment:
Since you use `$scope.model.account_id && $scope.model.account_id > 0` in more than one place, it would be good to assign it to a local variable.  Something like this:
```javascript
var hasSelectedAccount = $scope.model.account_id && $scope.model.account_id > 0;
$scope.mode = hasSelectedAccount ? 'selected' : 'all';
if (hasSelectedAccount) {
  // etc..
}
```